### PR TITLE
feat(ccls:1826) notifications sort

### PR DESCRIPTION
**Pull request by keith-h**

### CODE REVIEW

File: src/main/java/uk/gov/laa/ccms/caab/controller/ActionsAndNotificationsController.java [hunk 1]
```diff
@@ -41,13 +54,25 @@ public class ActionsAndNotificationsController {
   public String getNotifications(
       @RequestParam(value = "page", defaultValue = "0") int page,
       @RequestParam(value = "size", defaultValue = "10") int size,
```
Comment (PhilDigitalJustice) [inline src/main/java/uk/gov/laa/ccms/caab/controller/ActionsAndNotificationsController.java:56]: Do we want users to be able to amend the page size?

Comment (PhilDigitalJustice) [reply src/main/java/uk/gov/laa/ccms/caab/controller/ActionsAndNotificationsController.java:56]: we could just hardcode 10 couldn't we?

File: src/main/java/uk/gov/laa/ccms/caab/controller/ActionsAndNotificationsController.java [hunk 2]
```diff
@@ -41,13 +54,25 @@ public class ActionsAndNotificationsController {
...
-    criteria.setLoginId(user.getLoginId());
-    criteria.setUserType(user.getUserType());
-    //temporarily set the assigned to attribute until search notifications is implemented
-    criteria.setAssignedToUserId(user.getLoginId());
+
+    log.info("GET /notifications");
+
+    // first time the criteria has been set?
+    // set the basic attributes for a user search
+    if (!criteria.isInstantiated()) {
```
Comment (PhilDigitalJustice) [inline src/main/java/uk/gov/laa/ccms/caab/controller/ActionsAndNotificationsController.java:66]: Is this necessary?

File: src/main/resources/templates/notifications/actions-and-notifications.html [hunk 3]
```diff
@@ -31,28 +31,28 @@ <h1 class="govuk-heading-l">
...
-                <td class="govuk-table__cell" th:text="${#dates.format(notification.notificationDetail.assignDate,'dd-MM-yyyy')}"></td>
-                <td class="govuk-table__cell" th:text="${notification.notificationDetail.subject}"></td>
-                <td class="govuk-table__cell" th:text="${#dates.format(notification.notificationDetail.dueDate,'dd-MM-yyy')}"></td>
-                <td class="govuk-table__cell" th:text="${notification.user.userLoginId}"></td>
-                <td class="govuk-table__cell" th:text="${notification.notificationDetail.status}"></td>
-                <td class="govuk-table__cell" th:text="${notification.caseReferenceNumber}"></td>
-                <td class="govuk-table__cell" th:text="${notification.providerCaseReferenceNumber}"></td>
-                <td class="govuk-table__cell" th:text="${notification.clientName}"></td>
-                <td class="govuk-table__cell" th:text="${notification.feeEarner}"></td>
+                <td class="govuk-table__cell govuk-!-font-size-17" th:text="${#dates.format(notification.assignDate,'dd-MM-yyyy')}"></td>
```
Comment (PhilDigitalJustice) [inline src/main/resources/templates/notifications/actions-and-notifications.html:47]: Should it be slashed not hyphens?
The rest of PUI deisplays as dd/mm/yyyy

File: src/main/resources/templates/notifications/actions-and-notifications.html [hunk 4]
```diff
@@ -31,28 +31,28 @@ <h1 class="govuk-heading-l">
...
+                <td class="govuk-table__cell govuk-!-font-size-17" th:text="${notification.subject}"></td>
+                <td class="govuk-table__cell govuk-!-font-size-17" th:text="${#dates.format(notification.dueDate,'dd-MM-yyy')}"></td>
```
Comment (PhilDigitalJustice) [inline src/main/resources/templates/notifications/actions-and-notifications.html:49]: Should it be slashed not hyphens?
The rest of PUI deisplays as dd/mm/yyyy

File: src/main/resources/templates/notifications/partials/sorting.html
Comment (PhilDigitalJustice) [inline src/main/resources/templates/notifications/partials/sorting.html:1]: Could we compine paging and sorting into a single fragment html file 
then you can choose to pull in one or both?

File: src/main/java/uk/gov/laa/ccms/caab/controller/ActionsAndNotificationsController.java (hunk 2 above)
Comment (keith-h) [reply src/main/java/uk/gov/laa/ccms/caab/controller/ActionsAndNotificationsController.java:66]: Will put in session

File: src/main/resources/templates/notifications/actions-and-notifications.html (hunk 3 above)
Comment (keith-h) [reply src/main/resources/templates/notifications/actions-and-notifications.html:47]: Done

File: src/main/resources/templates/notifications/actions-and-notifications.html (hunk 4 above)
Comment (keith-h) [reply src/main/resources/templates/notifications/actions-and-notifications.html:49]: Done - and corrected the year format to `yyyy`

File: src/main/resources/templates/notifications/partials/sorting.html
Comment (keith-h) [reply src/main/resources/templates/notifications/partials/sorting.html:1]: I'll give it a bash

File: src/main/java/uk/gov/laa/ccms/caab/controller/ActionsAndNotificationsController.java (hunk 1 above)
Comment (keith-h) [reply src/main/java/uk/gov/laa/ccms/caab/controller/ActionsAndNotificationsController.java:61]: think we agreed to leave as is?

**Review by PhilDigitalJustice — APPROVED** [review]
LGTM

**Pull request merged**